test/CMakeLists.txt

```diff
@@ -29,14 +28,28 @@ endif()
 # this interface target is to collect
 # compile/link options for the test
 add_library(portsofcall_iface INTERFACE)
-if (PORTABILITY_STRATEGY_KOKKOS)
+if (PORTS_OF_CALL_TEST_PORTABILITY_STRATEGY STREQUAL "Kokkos")
   if(NOT TARGET Kokkos::kokkos)
-    find_package(Kokkos REQUIRED)
+    find_package(Kokkos QUIET)
+    if(NOT Kokkos_FOUND)
+      message(STATUS "Fetching Kokkos as needed")
+      include(FetchContent)
+      FetchContent_Declare(
+        Kokkos
+        GIT_REPOSITORY https://github.com/kokkos/kokkos.git
+        # This is most recent but any version that supports C++17 is
+        # fine
+        GIT_TAG 4.3.01)
+      FetchContent_MakeAvailable(Kokkos)
+    endif()
   endif()
 
   target_link_libraries(portsofcall_iface INTERFACE Kokkos::kokkos)
   # this comes with ports-of-call target
   target_compile_definitions(portsofcall_iface INTERFACE PORTABILITY_STRATEGY_KOKKOS)
+elseif (PORTS_OF_CALL_TEST_PORTABILITY_STRATEGY STREQUAL "Cuda")
+  message(FATAL_ERROR "Cuda tests not yet fully supported")
```

Comment:
CMake execution will stop here. So the line below will not be executed.

Reply:
Yes that's the intent for now.